java/23c-jdbc-boolean/src/main/java/com/oracle/dev/jdbc/QueryWithBooleanDataTypeColumn.java

```diff
@@ -0,0 +1,71 @@
+/*
+  Copyright (c) 2024, Oracle and/or its affiliates.
+
+  This software is dual-licensed to you under the Universal Permissive License
+  (UPL) 1.0 as shown at https://oss.oracle.com/licenses/upl or Apache License
+  2.0 as shown at http://www.apache.org/licenses/LICENSE-2.0. You may choose
+  either license.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+     https://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+*/
+
+package com.oracle.dev.jdbc;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+
+import oracle.jdbc.pool.OracleDataSource;
+
+public class QueryWithBooleanDataTypeColumn {
+
+	public static void main(String[] args) {
+
+		OracleDataSource ods = null;
+		Connection conn = null;
+
+		try {
+			
+			ods = new OracleDataSource();
+			
+			// jdbc:oracle:thin@[hostname]:[port]/[DB service/name]
+			ods.setURL("jdbc:oracle:thin:@localhost:1521/FREEPDB1");			
+			ods.setUser("[Username]");
+		    ods.setPassword("[Password]");	
+						
+			conn = ods.getConnection();
+			
+			PreparedStatement stmt = conn.prepareStatement("SELECT * FROM HQ_EMPLOYEE WHERE ACTIVE");						
+			//PreparedStatement stmt = conn.prepareStatement("SELECT * FROM HQ_EMPLOYEE WHERE ACTIVE = ?");	
+			//stmt.setBoolean(1, true);				
+			//stmt.setInt(1, 1);
+			//stmt.setString(1, "1");
+					
+			ResultSet rslt = stmt.executeQuery();
+			StringBuilder sb = new StringBuilder();
+			while (rslt.next()) {
+				sb.append(rslt.getInt(1)).append("|").append(rslt.getString(2)).append("|").append(rslt.getString(3))
+						.append("|").append(rslt.getBoolean(4)).append("\n");
```

Comment:
getBoolean is the only API that demonstrates how to use Boolean. To be more meaningful, the demo could also show the other techniques to bind a Boolean and retrieve a boolean value.

Reply:
@jeandelavarene as we talked about, the main purpose of this code sample (and the associated blog post) was to specifically explore "JDBC type compatibility" as the title informs, and that's the reason why it has 12 (twelve) different scenarios to present both the supported (compatible) ones and the unsupported (incompatible) ones. 

I talked with Kuassi and we'll create another blog post to provide a full-fledged use of the API with a single (compatible) scenario only.